### ISSUE TITLE
Unhandled Exception when exiting to desktop

### ISSUE THREAD

**Issue by jrmain** [EDDiscovery/EDDiscovery#1311]
"There was an unhandled UI exception. Please report this..."

Reporting as instructed. Requested log attached.

[Trace_20170928144508.0.log](https://github.com/EDDiscovery/EDDiscovery/files/1342525/Trace_20170928144508.0.log)

**Comment by phroggster** [comment]
```
[2017-09-28 15:06:43Z] 
==== UNHANDLED UI EXCEPTION ====
System.NullReferenceException: Object reference not set to an instance of an object.
   at ExtendedControls.PictureBoxHotspot.AddTextAutoSize(Point topleft, Size max, String label, Font fnt, Color c, Color backcolour, Single backscale, Object tag, String tiptext)
   at EDDiscovery.UserControls.UserControlShoppingList.Display()
   at System.Action`2.Invoke(T1 arg1, T2 arg2)
   at EDDiscovery.EDDiscoveryController.NewEntry(JournalEntry je)
   at EliteDangerousCore.EDJournalClass.ScanTickDone(List`1 entries)
==== cut ====
```

**Comment by phroggster** [comment]
The only way I could see this occurring is if `PictureBoxHotspot.elements` is `null`, which means that `UserControlShoppingList.pictureBoxList` has been disposed of, which means that `UserControlShoppingList` has been disposed. Ergo, `UserControlShoppingList.Dispose(bool)` apparently needs to also unsubscribe from the NewEntry event.

**Comment by iainross** [comment]
Should be fixed by #1324

**Comment by phroggster** [comment]
Resolved in (what should become v8.3.3) via cd3ca18ab0af9f8ac780fedbed5ba36488c565d8.
